### PR TITLE
Port moveit_core/utils to ROS 2

### PR DESCRIPTION
**Pull request by vmayoral**
Initial iteration was made on a different "master" status

### CODE REVIEW

**Comment by vmayoral** [comment]
See https://github.com/ros-planning/moveit2/pull/62#issuecomment-480720464

File: moveit_core/utils/CMakeLists.txt [hunk 1]
```diff
@@ -2,34 +2,37 @@ set(MOVEIT_LIB_NAME moveit_utils)
 
 add_library(${MOVEIT_LIB_NAME}
   src/lexical_casts.cpp
-  src/xmlrpc_casts.cpp
+  # src/xmlrpc_casts.cpp # TODO: solve the conflicts with xmlrpc dependencies
```
Comment (henningkayser) [inline moveit_core/utils/CMakeLists.txt]: I think `xmlrpc_casts.cpp` will not be used anymore since XML-RPC is not supported in ROS 2.
`parseDouble()` and `isArray()` should be redundant in the future but we have to check where`isStruct()` is used and if we need to reimplement it somehow.
Let's remove `xmlrpc_casts.cpp` and add TODOs everywhere where it is used.
These parts will need to be changed anyway since the same missing XML-RPC dependencies are used.

**Comment by ahcorde** [comment]
@henningkayser can you review it?

**Comment by ahcorde** [comment]
@henningkayser friendly ping

**Comment by henningkayser** [comment]
@ahcorde removing this files won't be enough since the functionality needs to be reimplemented eventually. Those files that `xmlrpc_casts.h`  will fail to compile which is just terrible to debug and could slow down the port.
* moveit_plugins/moveit_simple_controller_manager/src/follow_joint_trajectory_controller_handle.cpp
* moveit_plugins/moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp
* moveit_core/kinetmatic_constraints/src/utils.cpp
Please fix related includes and add a descriptions for all function calls that need to be replaced somehow. Ideally we would replace the implementation right now but I believe that we still need to have some further discussions on how we handle parameter config files.

**Comment by ahcorde** [comment]
we decided to remove `xmlrpc` because in ROS 2 is not used anymore. This concept of getting parameters with xmlrpc has disappeared. There are new ways the get parameters though ROS 2 and DDS. 

We have replaced all these calls in the code with the corresponding new ROS 2.0 code. I think it's better to discuss if we can include new abstractions for getting parameters than maintaining this old ROS code.

**Comment by henningkayser** [comment]
> we decided to remove `xmlrpc` because in ROS 2 is not used anymore. This concept of getting parameters with xmlrpc has disappeared. There are new ways the get parameters though ROS 2 and DDS.





> we decided to remove `xmlrpc` because in ROS 2 is not used anymore. This concept of getting parameters with xmlrpc has disappeared. There are new ways the get parameters though ROS 2 and DDS.
 
Yes, that's why I suggested to remove `xmlrpc`.

> We have replaced all these calls in the code with the corresponding new ROS 2.0 code. I think it's better to discuss if we can include new abstractions for getting parameters than maintaining this old ROS code.

I wasn't aware that this was already implemented. Is it still under review? If yes I would suggest adding related descriptions and would +1 this PR

**Comment by ahcorde** [comment]
You can open this (huge) [PR](https://github.com/AcutronicRobotics/moveit2/commit/1fc3cf4bb3d1979fe71ac6ff175c3667aa68ace7) and have a look to moveit_planners/ompl/ompl_interface/src/ompl_interface.cpp.  You will see the changes.

Most of these changes are still pending of been ported. But You will find a TODO (anasarrak) or (ahcorde). The parameters which were ported are the most relevant.

File: moveit_core/utils/include/moveit/utils/robot_model_test_utils.h [hunk 2]
```diff
@@ -182,10 +182,10 @@ class RobotModelBuilder
 
 private:
   /** \brief Adds different collision geometries to a link. */
-  void addLinkCollision(const std::string& link_name, const urdf::CollisionSharedPtr& coll, geometry_msgs::Pose origin);
+  void addLinkCollision(const std::string& link_name, urdf::CollisionSharedPtr coll, geometry_msgs::msg::Pose origin);
```
Comment (mlautman) [inline moveit_core/utils/include/moveit/utils/robot_model_test_utils.h]: Why were these changed from const references to copies if the shared pointer? The code hasn't changed at all. Why should these?

Comment (ahcorde) [reply moveit_core/utils/include/moveit/utils/robot_model_test_utils.h]: my bad. Fixed
